src/commands/run.rs

```diff
@@ -643,7 +643,8 @@ impl RunCommand {
                         bail!("Cannot enable wasi-http for core wasm modules");
                     }
                     CliLinker::Component(linker) => {
-                        wasmtime_wasi_http::proxy::add_to_linker(linker)?;
+                        linker.allow_shadowing(true);
+                        wasmtime_wasi_http::proxy::sync::add_to_linker(linker)?;
```

Comment:
I think it would be best if `allow_shadowing` was disabled because that's otherwise easy to accidentally run into issues.

There's unfortunately not a great solution for the error you're running into here, so in the meantime I think it would be best to expose individual `add_to_linker` functions per-interface and only invoke the wasi-http-related ones here.

Reply:
done. as it is only temporary, i exposed function only in sync module and kept out of docs.